plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/IdentityPinotHostMapper.java

```diff
@@ -32,4 +33,11 @@ public ServerInstance getServerInstance(String serverHost)
     {
         return new ServerInstance(InstanceConfig.toInstanceConfig(serverHost));
     }
+
+    @Override
+    public HostAndPort getServerGrpcHostAndPort(String serverHost, int grpcPort)
+    {
+        ServerInstance serverInstance = getServerInstance(serverHost);
+        return HostAndPort.fromParts(serverInstance.getHostname(), grpcPort);
```

Comment:
Actually `ServerInstance` has an API to get grpcPort, so you don't need to pass it. 
https://github.com/apache/pinot/blob/master/pinot-core/src/main/java/org/apache/pinot/core/transport/ServerInstance.java#L93

This API can just be: `public HostAndPort getServerGrpcHostAndPort(String serverHost)`

In this case, you don't even need to have the config for grpcPort, just make this best effort try. If the grpc port is -1, you return null here, and the query will use netty query endpoint. Once server has grpc, then the query will use grpc query endpoint.

Reply:
I tried this connecting to a real cluster and the server instance returns `-1` for the grpc port, it did not work. Is it ok to keep it explicitly specified? lmk what you think @xiangfu0 .

Reply:
You mean grpc is enabled but serverInstance gives grpc port -1? If that is the case, please keep this grpc port config as the backup and use the grpc port that > 0 if from the ServerInstance. 
So the logic is best effort from ServerInstance if possible or use grpc port if not provided.